src/engine/GroupBy.cpp

```diff
@@ -891,25 +885,24 @@ void GroupBy::substituteAllAggregates(
     sparqlExpression::EvaluationContext& evaluationContext,
     const ad_utility::HashMapWithMemoryLimit<KeyType, ValueType<numAggregates>>&
         map,
-    IdTable* groupValues) {
-  // Multiple aggregates, need to substitute them away and then finally
-  // evaluate the top-level expression.
+    IdTable* resultTable) {
+  // Substitute in the results of all aggregates of `info`.
   for (auto& aggregate : info) {
     sparqlExpression::VectorWithMemoryLimit<ValueId> aggregateResults(
         getExecutionContext()->getAllocator());
 
     // Get all aggregate results as a vector
+    decltype(auto) groupValues = resultTable->getColumn(0);
```

Comment:
Both of those can be refactored using `std::ranges::transform` etc. Remember this and contact me before your next PR, so that I can show you how.